### PR TITLE
Ignore case while detecting ReqUUID

### PR DESCRIPTION
**Pull request by sean1006**
Previously, case sensitivity could cause mis-matches in detecting the presence of a ReqUUID which ends up in Zanzibar populating an additional ReqUUID. This addresses the ReqUUID issue but does not make headers in the tcahnnel_server fully case insensitive, as there are further use cases which must be addressed first.

### CODE REVIEW

File: runtime/tchannel_server.go [hunk 1]
```diff
@@ -206,7 +206,13 @@ func (s *TChannelRouter) handleHeader(
 		return ctx, err
 	}
 
-	reqUUID, ok := c.reqHeaders[s.requestUUIDHeaderKey]
+	// TODO: Make headers case insensitive. Remove instances of headers being used in a case
```
Comment (rpatali) [inline runtime/tchannel_server.go]: can you write a test?

File: runtime/tchannel_server.go [hunk 2]
```diff
@@ -206,7 +206,13 @@ func (s *TChannelRouter) handleHeader(
 		return ctx, err
 	}
 
-	reqUUID, ok := c.reqHeaders[s.requestUUIDHeaderKey]
```
Comment (rpatali) [inline runtime/tchannel_server.go:209]: can you simply not use:
```
reqUUID, ok := c.reqHeaders[textproto.CanonicalMIMEHeaderKey(s.requestUUIDHeaderKey)]
```

Comment (sean1006) [reply runtime/tchannel_server.go:209]: unfortunately, we need to standardize input map keys

File: runtime/tchannel_server.go (hunk 1 above)
Comment (sean1006) [reply runtime/tchannel_server.go]: will update

File: runtime/tchannel_server.go [hunk 3]
```diff
@@ -300,3 +297,18 @@ func DecorateWithRecover(
...
+// uuid if no such UUID exists in request headers
+func getRequestUUID(
+	reqUUIDHeaderKey string,
+	reqHeaders map[string]string
+) string {
+	canonicalHeaderKey := textproto.CanonicalMIMEHeaderKey(s.requestUUIDHeaderKey)
+	for key, value := range reqHeaders {
+		if key == canonicalHeaderKey && value != "" {
+			return value
+		}
```
Comment (rpatali) [inline runtime/tchannel_server.go]: `reqHeaders` is a map :)

File: runtime/tchannel_server_test.go [hunk 4]
```diff
@@ -93,3 +94,44 @@ func (m mockRWTStruct) ToWire() (wire.Value, error) {
...
+
+func TestGetRequestUUID(t *testing.T) {
+	uuidHeaderKey := "test-uuid-key"
+	testUuidValue := uuid.New()
+
+	testCases := map[string]struct {
+		reqHeaders 		map[string]string
+		expectedEquals  string
+	}{
+		"Success: return new UUID when no request header map is given": {},
```
Comment (rpatali) [inline runtime/tchannel_server_test.go:106]: when null header map is given as compared to empty map given in the next case is better naming

**Review by rpatali — APPROVED** [review]
LGTM

**Pull request merged**